lib/Backend/GlobOpt.cpp

```diff
@@ -19069,6 +19107,30 @@ GlobOpt::OptHoistInvariant(
     IR::BailOutKind bailoutKind)
 {
     BasicBlock *landingPad = loop->landingPad;
+
+    IR::Opnd* src1 = instr->GetSrc1();
+    if (src1)
+    {
+        // We are hoisting this instruction possibly past other uses, which might invalidate the last use info. Clear it.
+        OptHoistToLandingPadUpdateValueType(landingPad, instr, src1, src1Val);
```

Comment:
Actually, how does updating the value type on the operand in the hoisted instruction help? The hoisted instr won't be processed by the forward pass, correct? Is it that the deadstore pass takes some unwanted action if the value type is left unchanged?